src/cachemap/proxies/redis/index.ts

```diff
@@ -18,49 +18,73 @@ export default class RedisProxy {
   }
 
   public async clear(): Promise<void> {
-    await new Promise((resolve) => {
+    await new Promise((resolve, reject) => {
       this._client.flushdb((err, reply) => {
-        resolve();
+        if (err) {
+          reject(err);
+        } else {
+          resolve();
+        }
       });
     });
   }
 
   public async delete(key: string): Promise<boolean> {
-    return new Promise((resolve: (value: boolean) => void) => {
+    return new Promise((resolve: (value: boolean) => void, reject) => {
       this._client.del(key, (err, reply) => {
-        resolve(!!reply);
+        if (err) {
+          reject(err);
+        } else {
+          resolve(!!reply);
+        }
       });
     });
   }
 
   public async get(key: string): Promise<any> {
-    return new Promise((resolve: (value: any) => void) => {
+    return new Promise((resolve: (value: any) => void, reject) => {
       this._client.get(key, (err, reply) => {
-        resolve(JSON.parse(reply));
+        if (err) {
+          reject(err);
+        } else {
+          resolve(JSON.parse(reply));
+        }
       });
     });
   }
 
   public async has(key: string): Promise<boolean> {
-    return new Promise((resolve: (value: boolean) => void) => {
+    return new Promise((resolve: (value: boolean) => void, reject) => {
       this._client.exists(key, (err, reply) => {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this function so that its implementation is not identical to the one on line 34. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4144)